src/quail/SensorTask.cpp

```diff
@@ -0,0 +1,41 @@
+#include "PressureSensor.hpp"
+#include "LoadSensor.hpp"
+#include "ThermalSensor.hpp"
+
+#include "SensorTask.hpp"
+
+SensorTask::SensorTask(uint8_t priority) : Task(priority, "Sensor"){};
+
+Sensor SensorTask::sensors[NUM_SENSORS] = 
+{
+    PressureSensor(0, "PT1", Ad7124::AIN1Input,PressureSensor::RANGE_1000) // channel, ADC in port, PT range
+};
+
+void SensorTask::activity()
+{
+    while(true)
+    {
+        adcdata_t adcpacket;
+        adcbuf.receive(adcpacket, NEVER); // wait until an adc packet is ready, state goes BLOCKED until not empty
+        Sensor cur_sensor = sensors[adcpacket.channel]; // sensor for which we just received data
+        float val = cur_sensor.convertToFloat(adcpacket.dataword); // convert ADC packet from 24bit to float
+        sensordata_t sensedata;
+        sensedata.ch_name = cur_sensor.ch_name;
+        sensedata.value = val;
+        sys.tasks.statetask.updateSensorState(sensedata);
+        Serial.println(cur_sensor.ch_name.c_str());
+        Serial.println(val, 5);
+    }
+};
+
+void SensorTask::initialize()
```

Comment:
This is a bit odd in my mind, because this is a function of SensorTask, but it called from ADCTask. I can see why, but I'm gonna ponder it.

Reply:
yeah I didn't have a better solution, since the SensorTask owns the sensor list... lmk if you come up with something cleaner.